src/server/listeners/constellationListener.js

```diff
@@ -15,23 +15,33 @@ class ConstellationListener {
   }
 
   async _onTransaction(constellationTransaction) {
-    console.log('Got transaction from Constellation API');
     if (!constellationTransaction.isSignedBySourceAccount()) {
       return;
     }
 
     try {
       const transaction = new Transaction({
         xdr: constellationTransaction.xdr,
-        externalId: constellationTransaction.externalId,
-        submittedFrom: 'constellation'
+        externalId: constellationTransaction.id,
```

Comment:
These listeners are very similar... should figure out a way to refactor their commonalities.